frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -624,8 +608,13 @@ export const funnelLogic = kea<funnelLogicType>({
             // If user started from empty state (<2 steps) and added a new step
             const shouldRefresh =
                 values.filters?.events?.length === 2 && values.lastAppliedFilters?.events?.length === 1
+            // If layout is the only thing that changes
+            const shouldNotRefresh = equal(
+                Object.assign({}, values.filters, { layout: undefined }),
+                Object.assign({}, values.lastAppliedFilters, { layout: undefined })
+            )
 
-            if (refresh || shouldRefresh || clickhouseFeaturesEnabled) {
+            if ((refresh || shouldRefresh || clickhouseFeaturesEnabled) && !shouldNotRefresh) {
```

Comment:
This line feels a bit messy. `shouldRefresh && !shouldNotRefresh` :D.

Perhaps rename the double negation var to `onlyLayoutChanged`?